Source/SequenceMethodCall.cs

```diff
@@ -0,0 +1,97 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Linq.Expressions;
+using System.Reflection;
+
+using Moq.Proxy;
+
+namespace Moq
+{
+	// this corresponds to a setup created by `mock.SetupSequence`
+	internal class SequenceMethodCall : MethodCall
+	{
+		// contains the responses set up with `CallBase`, `Pass`, `Returns`, and `Throws`
+		// TODO: the use of a queue without keeping track of the number of received calls, and tying each
+		// response to an expected invocation count, means that things won't work quite like they used to!
+		// this is possibly a breaking change right now.
+		private ConcurrentQueue<Tuple<ResponseKind, object>> responses;
+
+		public SequenceMethodCall(Mock mock, Condition condition, Expression originalExpression, MethodInfo method, params Expression[] arguments)
```

Comment:
Not sure if we need all these parameters, e.g. `condition`. Could probably be shortened.